src/components/MessagesList/MessagesGroup/Message/MessagePart/AudioPlayer.vue

```diff
@@ -0,0 +1,65 @@
+<!--
+  - @copyright Copyright (c) 2021 Marco Ambrosini <marcoambrosini@pm.me>
+  -
+  - @author Marco Ambrosini <marcoambrosini@pm.me>
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+-->
+
+<template>
+	<div class="wrapper">
+		<audio
+			controls
+			:src="davURL">
+			Your browser does not support the
+			<code>audio</code> element.
+		</audio>
+	</div>
+</template>
+
+<script>
+import { generateRemoteUrl } from '@nextcloud/router'
+import { getCurrentUser } from '@nextcloud/auth'
+
+export default {
+	name: 'AudioPlayer',
+
+	props: {
+		/**
+		 * File name
+		 */
+		name: {
+			type: String,
+			required: true,
+		},
+		/**
+		 * File path relative to the user's home storage,
+		 * or link share root, includes the file name.
+		 */
+		path: {
+			type: String,
+			required: true,
+		},
+	},
+
+	computed: {
+		davURL() {
+			return generateRemoteUrl(`dav/files/${getCurrentUser().uid}/`) + this.path
```

Comment:
this won't work with special chars in path/file names, see https://github.com/nextcloud/spreed/issues/5672

also, is this restricted to logged in users or should guests also be able to access audio files ? if yes, might need to use the public download link like in https://github.com/nextcloud/spreed/blob/master/src/components/MessagesList/MessagesGroup/Message/MessagePart/FilePreview.vue#L292

Reply:
👍